py/dml/structure.py

```diff
@@ -1895,12 +1902,18 @@ def mkobj2(obj, obj_specs, params, each_stmts):
                     else:
                         report(e)
         if (dml.globals.dml_version != (1, 2)
-            or not dml.globals.compat_dml12):
+            or compat.dml12_misc not in dml.globals.enabled_compat):
             # TODO: this should be handled cleaner in the case of pure
             # 1.4 code
             for p in obj.get_components():
                 sym = global_scope.lookup(p.name)
-                if sym:
+                if sym and (
+                        # hacky workaround for the ExpressionSymbol
+                        # implicitly added above. Needed when importing 1.4
+                        # code from 1.2 with --no-compat=dml12_misc
+                        dml.globals.dml_version != (1, 2)
+                        or p.site.dml_version == (1, 2)
+                        or p.site != sym.site):
```

Comment:
OW
By the way, should we have some standard way to annotate hacks and todos? I've always used `# HACK:` and `# TODO: ` because my editor highlights them.